Bugfix/create course/fix save button lock issue1334

anditurdiu opened the pull request:
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes

### Motivation and Context
This pull request was opened as a fix to Issue #1334

### Description
The `HttpErrorResponse` from the server has an empty `error field`. Instead the error was reflected in the header of the response as `x-artemisapp-alert`. This caused a `NullPointerException` which did not allow the "saving" state to be completed thereby locking the save button. As a fix I introduced a null test for `HttpErrorResponse.error`. When it is null, it fetches the error Message from the header `x-artemisapp-alert`.

### Steps for Testing
1. Log in to Artemis
2. Navigate to Course Administration
3. Create a course with an already used shortname. E.g. ATC
4. Hit save

### Screenshots
![image](https://user-images.githubusercontent.com/15110960/80690373-ffef7280-8ace-11ea-8e31-48c009d4c9cc.png)

krusche commented:
@anditurdiu could you please fix the 3 warnings shown by GitHub in the Files changed tab? I know 2 of them were not introduced by you, but then we have fixed them as well. Thanks

anditurdiu commented:
> @anditurdiu could you please fix the 3 warnings shown by GitHub in the Files changed tab? I know 2 of them were not introduced by you, but then we have fixed them as well. Thanks

I removed all usused imports referenced by github

krusche commented:
Thanks 👍🏻
